### PR TITLE
Fixed querying ads over HTTPS

### LINKED ISSUE #352: Using TLS/HTTP2 breaks Query ad list
@C0rn3j commented on [Fri Jan 06 2017](https://github.com/pi-hole/pi-hole/issues/1100)

Using HTTP works fine as intended.

I am not certain what exactly breaks this(but it is HTTPS).

Package versions:
nginx-mainline 1.11.8-1
linux-raspberrypi 4.4.39-1
php-fpm 7.0.14-1
pi-hole-server 2.11.1-1

As you can see I'm using nginx and not lighttpd.

Logs - some of them might not be relevant as I've been trying to disable/enable HTTP2
http://pastebin.com/P5G0MDSL

Relevant part of the nginx config(ignore that 443 is commented out, it is normally enabled): 
http://pastebin.com/4fffPRy7

HTTP: http://i.imgur.com/aq9hKR4.png
HTTPS: http://i.imgur.com/FKwtIdV.png

---

    [X] I have read and understood the contributors guide.
    [X] The issue I am reporting can be replicated
    [X] The issue I'm reporting isn't a duplicate (see FAQs, closed issues, and open issues).

    How familiar are you with the codebase?: 1

---

@PromoFaux commented on [Sat Jan 07 2017](https://github.com/pi-hole/pi-hole/issues/1100#issuecomment-271035574)

Please ensure you follow the [issue template](https://github.com/pi-hole/pi-hole/blob/master/.github/ISSUE_TEMPLATE.md) when submitting a new issue. It's there to help us to help you!

---

@C0rn3j commented on [Sat Jan 07 2017](https://github.com/pi-hole/pi-hole/issues/1100#issuecomment-271103353)

I've added the checklist to the bottom of the issue.

---

@DL6ER commented on [Thu Jan 12 2017](https://github.com/pi-hole/pi-hole/issues/1100#issuecomment-272107867)

[First result on Google](http://serverfault.com/questions/517190/nginx-1-fastcgi-sent-in-stderr-primary-script-unknown) for the error shown in your log:

> The error message “primary script unknown” is **always** related to a wrongly set SCRIPT_FILENAME in the nginx fastcgi_param directive.

I don't even see this directive in your config.

Also, you provided some other `[info]` message was are a few minutes after the `[error]` messages - what action did you do at this time or was it completely unrelated after all?

---

@C0rn3j commented on [Thu Jan 12 2017](https://github.com/pi-hole/pi-hole/issues/1100#issuecomment-272120351)

I fixed it myself, wanted to do a pull request but can't find anywhere in the github repo where 
admin/scripts/pi-hole/js/ folder is located, could someone point me in the right direction?

### PR DESCRIPTION
**Pull request by C0rn3j**
**By submitting this pull request, I confirm the following (please check boxes, eg [X] - no spaces) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

1

---
changed http:// to // to always use the correct protocol

Fixes #352

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

File: scripts/pi-hole/js/queryads.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ function eventsource() {
     }
 
     var host = window.location.host;
-    var source = new EventSource("http://"+host+"/admin/scripts/pi-hole/php/queryads.php?domain="+domain.val().toLowerCase()+"&"+exact);
+    var source = new EventSource("//"+host+"/admin/scripts/pi-hole/php/queryads.php?domain="+domain.val().toLowerCase()+"&"+exact);
```
Comment (DL6ER) [inline scripts/pi-hole/js/queryads.js]: Please change it to

```
"scripts/pi-hole/php/queryads.php?domain=..."
```

Comment (Mcat12) [inline scripts/pi-hole/js/queryads.js]: That change causes the block page to not work, as it tries to find the script at http://<some_ad_domain>/scripts/...
```
EventSource's response has a MIME type ("text/html") that is not "text/event-stream". Aborting the connection.
```

File: scripts/pi-hole/js/queryads.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ function eventsource() {
     }
 
     var host = window.location.host;
-    var source = new EventSource("http://"+host+"/admin/scripts/pi-hole/php/queryads.php?domain="+domain.val().toLowerCase()+"&"+exact);
+    var source = new EventSource("scripts/pi-hole/php/queryads.php?domain="+domain.val().toLowerCase()+"&"+exact);
```
Comment (Mcat12) [inline scripts/pi-hole/js/queryads.js]: Change back to the `//"+host+"/admin/` you had before

**Comment by Mcat12** [comment]
Oh, I just edited my comment to ask you to instead change it to `"admin/scripts/...`, sorry!

**Comment by C0rn3j** [comment]
Hopefully it doesn't need any more changes now :P

**Comment by DL6ER** [comment]
:shipit:

**Pull request merged**